---Import Necessary Modules---

In [2]:
import pandas as pd
from glob import glob
import os
import numpy as np

In [4]:
csv_files = glob("/Users/elliotevins/movie-project/movie-analysis-project/unzipped-data/*.csv")
csv_files

['/Users/elliotevins/movie-project/movie-analysis-project/unzipped-data/tmdb.movies.csv',
 '/Users/elliotevins/movie-project/movie-analysis-project/unzipped-data/imdb.title.crew.csv',
 '/Users/elliotevins/movie-project/movie-analysis-project/unzipped-data/tn.movie_budgets.csv',
 '/Users/elliotevins/movie-project/movie-analysis-project/unzipped-data/imdb.title.ratings.csv',
 '/Users/elliotevins/movie-project/movie-analysis-project/unzipped-data/imdb.name.basics.csv',
 '/Users/elliotevins/movie-project/movie-analysis-project/unzipped-data/imdb.title.principals.csv',
 '/Users/elliotevins/movie-project/movie-analysis-project/unzipped-data/imdb.title.akas.csv',
 '/Users/elliotevins/movie-project/movie-analysis-project/unzipped-data/bom.movie_gross.csv',
 '/Users/elliotevins/movie-project/movie-analysis-project/unzipped-data/imdb.title.basics.csv']

In [5]:
d = {}
for file in csv_files:
    d[file] = pd.read_csv(file)

In [6]:
d.keys()

dict_keys(['/Users/elliotevins/movie-project/movie-analysis-project/unzipped-data/tmdb.movies.csv', '/Users/elliotevins/movie-project/movie-analysis-project/unzipped-data/imdb.title.crew.csv', '/Users/elliotevins/movie-project/movie-analysis-project/unzipped-data/tn.movie_budgets.csv', '/Users/elliotevins/movie-project/movie-analysis-project/unzipped-data/imdb.title.ratings.csv', '/Users/elliotevins/movie-project/movie-analysis-project/unzipped-data/imdb.name.basics.csv', '/Users/elliotevins/movie-project/movie-analysis-project/unzipped-data/imdb.title.principals.csv', '/Users/elliotevins/movie-project/movie-analysis-project/unzipped-data/imdb.title.akas.csv', '/Users/elliotevins/movie-project/movie-analysis-project/unzipped-data/bom.movie_gross.csv', '/Users/elliotevins/movie-project/movie-analysis-project/unzipped-data/imdb.title.basics.csv'])

In [7]:
csv_files_dict = {}
for filename in csv_files:
    filename_cleaned = os.path.basename(filename).replace(".csv", "").replace(".","_") #cleaning the file names
    filename_df = pd.read_csv(filename, index_col = 0)
    csv_files_dict[filename_cleaned] = filename_df

In [8]:
tmdb_movies_df = csv_files_dict['tmdb_movies']
imdb_title_crew_df = csv_files_dict['imdb_title_crew']
tn_movie_budgets_df = csv_files_dict['tn_movie_budgets']
imdb_title_ratings_df = csv_files_dict['imdb_title_ratings']
imdb_name_basics_df = csv_files_dict['imdb_name_basics']
imdb_title_principals_df = csv_files_dict['imdb_title_principals']
imdb_title_akas_df = csv_files_dict['imdb_title_akas']
bom_movie_gross_df = csv_files_dict['bom_movie_gross']
imdb_title_basics_df = csv_files_dict['imdb_title_basics']

In [9]:
# tmdb_movies_df.head()
# imdb_title_akas_df.head()
# imdb_title_crew_df.head()
# tn_movie_budgets_df.info()
# imdb_title_ratings_df.head()
# imdb_title_principals_df.index
# imdb_name_basics_df.head()
# bom_movie_gross_df.head()
# imdb_title_basics_df.head()

In [10]:
#import sqlite3

In [11]:
# conn = sqlite3.connect("movies_db.sqlite")
# #cur = conn.cursor()

In [12]:
# def create_sql_table_from_df(df, name, conn):
#     try:
#         df.to_sql(name,conn)
#         print(f'Created table {name}')
    
#     except Exception as e:
#         print(f'could not make table {name}')
#         print(e)

In [13]:
# for name, table in csv_files_dict.items():
#     create_sql_table_from_df(table, name, conn)

In [14]:
for item in csv_files_dict.items():
    print("================NEW DF================")
    print(item[0])
    print(item[1].info())
    print("================END DF================")

================NEW DF================
tmdb_movies
<class 'pandas.core.frame.DataFrame'>
Int64Index: 26517 entries, 0 to 26516
Data columns (total 9 columns):
genre_ids            26517 non-null object
id                   26517 non-null int64
original_language    26517 non-null object
original_title       26517 non-null object
popularity           26517 non-null float64
release_date         26517 non-null object
title                26517 non-null object
vote_average         26517 non-null float64
vote_count           26517 non-null int64
dtypes: float64(2), int64(2), object(5)
memory usage: 2.0+ MB
None
================END DF================
================NEW DF================
imdb_title_crew
<class 'pandas.core.frame.DataFrame'>
Index: 146144 entries, tt0285252 to tt9010172
Data columns (total 2 columns):
directors    140417 non-null object
writers      110261 non-null object
dtypes: object(2)
memory usage: 3.3+ MB
None
================END DF================
================NEW D

In [15]:
for item in csv_files_dict.items():
    print("================NEW DF================")
    print(item[0])
    print(item[1].head())
    print("================END DF================")

================NEW DF================
tmdb_movies
             genre_ids     id original_language  \
0      [12, 14, 10751]  12444                en   
1  [14, 12, 16, 10751]  10191                en   
2        [12, 28, 878]  10138                en   
3      [16, 35, 10751]    862                en   
4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Tra

Each data frame and its attributes are as follows:

![title](database_outline.png)

From here we can merge the tables with common attributes to simplify cleaning. First we'll join the tables that contain imdb's "tconst" index and a one to one relationship - that ends up being the title_basics_df and the title_ratings_df.

In [16]:
#imdb_title_crew_df.info()
#imdb_title_ratings_df.info()
imdb_title_principals_df.info()
# imdb_title_basics_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1028186 entries, tt0111414 to tt9692684
Data columns (total 5 columns):
ordering      1028186 non-null int64
nconst        1028186 non-null object
category      1028186 non-null object
job           177684 non-null object
characters    393360 non-null object
dtypes: int64(1), object(4)
memory usage: 47.1+ MB


Since title pincipals has the most entries (1,028,186) we'll use that as our foundation. Title basics and title crew have the same amount of entries so let's see which produces more from an inner join. 

In [17]:
imdb_title_basics_ratings = imdb_title_basics_df.join(imdb_title_ratings_df, how='left')
imdb_title_basics_ratings.info()


<class 'pandas.core.frame.DataFrame'>
Index: 146144 entries, tt0063540 to tt9916754
Data columns (total 7 columns):
primary_title      146144 non-null object
original_title     146123 non-null object
start_year         146144 non-null int64
runtime_minutes    114405 non-null float64
genres             140736 non-null object
averagerating      73856 non-null float64
numvotes           73856 non-null float64
dtypes: float64(3), int64(1), object(3)
memory usage: 13.9+ MB


In [18]:
#Check for n/a values
imdb_title_basics_ratings.isna().sum()

primary_title          0
original_title        21
start_year             0
runtime_minutes    31739
genres              5408
averagerating      72288
numvotes           72288
dtype: int64

Now we see that the imdb_crew_df has the same number of entries as our new df so let's merge those.

In [96]:
imdb_basics_crew_ratings = imdb_title_basics_ratings.join(imdb_title_crew_df, how='outer')
imdb_basics_crew_ratings.info()

<class 'pandas.core.frame.DataFrame'>
Index: 146144 entries, tt0063540 to tt9916754
Data columns (total 9 columns):
primary_title      146144 non-null object
original_title     146123 non-null object
start_year         146144 non-null int64
runtime_minutes    114405 non-null float64
genres             140736 non-null object
averagerating      73856 non-null float64
numvotes           73856 non-null float64
directors          140417 non-null object
writers            110261 non-null object
dtypes: float64(3), int64(1), object(5)
memory usage: 11.1+ MB


In [97]:
#Let's check to make sure the index values are unique and don't have any duplicates
len(imdb_basics_crew_ratings.index.unique())

146144

Great! Now it seems like we'll want to split up some of the values in the genres, writers, and directors columns:

In [98]:
#genre is a list of upto 3 genres, splitting it to get the individual genres
imdb_basics_crew_ratings['genres'] = imdb_basics_crew_ratings['genres'].fillna('n/a')
imdb_basics_crew_ratings['genres'] = imdb_basics_crew_ratings['genres'].apply(lambda x: x.split(",") if x else x)
imdb_basics_crew_ratings['genres'].unique

<bound method Series.unique of tconst
tt0063540      [Action, Crime, Drama]
tt0066787          [Biography, Drama]
tt0069049                     [Drama]
tt0069204             [Comedy, Drama]
tt0100275    [Comedy, Drama, Fantasy]
                       ...           
tt9916538                     [Drama]
tt9916622               [Documentary]
tt9916706                    [Comedy]
tt9916730                       [n/a]
tt9916754               [Documentary]
Name: genres, Length: 146144, dtype: object>

In [99]:
#making a set of all genres we have. set doesn't allow duplicate values

all_genres = set()
for genres in imdb_basics_crew_ratings['genres']:
    if genres:
        all_genres.update(genres)

In [100]:
all_genres

{'Action',
 'Adult',
 'Adventure',
 'Animation',
 'Biography',
 'Comedy',
 'Crime',
 'Documentary',
 'Drama',
 'Family',
 'Fantasy',
 'Game-Show',
 'History',
 'Horror',
 'Music',
 'Musical',
 'Mystery',
 'News',
 'Reality-TV',
 'Romance',
 'Sci-Fi',
 'Short',
 'Sport',
 'Talk-Show',
 'Thriller',
 'War',
 'Western',
 'n/a'}

In [101]:
#adding cols with zeros for all the genres we have. Will modify genre to 1 if the film is of that genre.

for genre in all_genres:
    imdb_basics_crew_ratings[genre] = np.zeros(shape=imdb_basics_crew_ratings.shape[0])
                                
imdb_basics_crew_ratings.head()

primary_title              original_title  \
tconst                                                                   
tt0063540                        Sunghursh                   Sunghursh   
tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
tt0069049       The Other Side of the Wind  The Other Side of the Wind   
tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
tt0100275         The Wandering Soap Opera       La Telenovela Errante   

           start_year  runtime_minutes                    genres  \
tconst                                                             
tt0063540        2013            175.0    [Action, Crime, Drama]   
tt0066787        2019            114.0        [Biography, Drama]   
tt0069049        2018            122.0                   [Drama]   
tt0069204        2018              NaN           [Comedy, Drama]   
tt0100275        2017             80.0  [Comedy, Drama, Fantasy]   

           averagerating  numvotes            directors  \
tconst                                                    
tt0063540            7.0      77.0            nm0712540   
tt0066787            7.2      43.0            nm0002411   
tt0069049            6.9    4517.0            nm0000080   
tt0069204            6.1      13.0            nm0611531   
tt0100275            6.5     119.0  nm0765384,nm0749914   

                                           writers  Crime  ...  Music  Action  \
tconst                                                     ...                  
tt0063540  nm0023551,nm1194313,nm0347899,nm1391276    0.0  ...    0.0     0.0   
tt0066787                                      NaN    0.0  ...    0.0     0.0   
tt0069049                      nm0000080,nm0462648    0.0  ...    0.0     0.0   
tt0069204                                nm0347899    0.0  ...    0.0     0.0   
tt0100275                      nm1360635,nm0749914    0.0  ...    0.0     0.0   

           Musical  Western  Game-Show  Mystery  Sport  Comedy  Documentary  \
tconst                                                                        
tt0063540      0.0      0.0        0.0      0.0    0.0     0.0          0.0   
tt0066787      0.0      0.0        0.0      0.0    0.0     0.0          0.0   
tt0069049      0.0      0.0        0.0      0.0    0.0     0.0          0.0   
tt0069204      0.0      0.0        0.0      0.0    0.0     0.0          0.0   
tt0100275      0.0      0.0        0.0      0.0    0.0     0.0          0.0   

           Adult  
tconst            
tt0063540    0.0  
tt0066787    0.0  
tt0069049    0.0  
tt0069204    0.0  
tt0100275    0.0  

[5 rows x 37 columns]

In [102]:
#setting the genre to be 1 if the film is of that genre

for index, row in imdb_basics_crew_ratings.iterrows():
    if row['genres']:
        for genre in row['genres']:
            imdb_basics_crew_ratings.loc[index, genre] = 1

imdb_basics_crew_ratings.head()

primary_title              original_title  \
tconst                                                                   
tt0063540                        Sunghursh                   Sunghursh   
tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
tt0069049       The Other Side of the Wind  The Other Side of the Wind   
tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
tt0100275         The Wandering Soap Opera       La Telenovela Errante   

           start_year  runtime_minutes                    genres  \
tconst                                                             
tt0063540        2013            175.0    [Action, Crime, Drama]   
tt0066787        2019            114.0        [Biography, Drama]   
tt0069049        2018            122.0                   [Drama]   
tt0069204        2018              NaN           [Comedy, Drama]   
tt0100275        2017             80.0  [Comedy, Drama, Fantasy]   

           averagerating  numvotes            directors  \
tconst                                                    
tt0063540            7.0      77.0            nm0712540   
tt0066787            7.2      43.0            nm0002411   
tt0069049            6.9    4517.0            nm0000080   
tt0069204            6.1      13.0            nm0611531   
tt0100275            6.5     119.0  nm0765384,nm0749914   

                                           writers  Crime  ...  Music  Action  \
tconst                                                     ...                  
tt0063540  nm0023551,nm1194313,nm0347899,nm1391276    1.0  ...    0.0     1.0   
tt0066787                                      NaN    0.0  ...    0.0     0.0   
tt0069049                      nm0000080,nm0462648    0.0  ...    0.0     0.0   
tt0069204                                nm0347899    0.0  ...    0.0     0.0   
tt0100275                      nm1360635,nm0749914    0.0  ...    0.0     0.0   

           Musical  Western  Game-Show  Mystery  Sport  Comedy  Documentary  \
tconst                                                                        
tt0063540      0.0      0.0        0.0      0.0    0.0     0.0          0.0   
tt0066787      0.0      0.0        0.0      0.0    0.0     0.0          0.0   
tt0069049      0.0      0.0        0.0      0.0    0.0     0.0          0.0   
tt0069204      0.0      0.0        0.0      0.0    0.0     1.0          0.0   
tt0100275      0.0      0.0        0.0      0.0    0.0     1.0          0.0   

           Adult  
tconst            
tt0063540    0.0  
tt0066787    0.0  
tt0069049    0.0  
tt0069204    0.0  
tt0100275    0.0  

[5 rows x 37 columns]

In [103]:
movie_details_df_modified = imdb_basics_crew_ratings.drop(columns = 'genres')

In [104]:

#checking the counts for all different genres

for col in movie_details_df_modified:
    print(f'Viewing values in col: {col}')
    print(f'Top 5 values:\n{movie_details_df_modified[col].value_counts()}')

Viewing values in col: primary_title
Top 5 values:
Home                  24
Broken                20
The Return            20
Alone                 16
Homecoming            16
                      ..
Brave New Jersey       1
Haiduc                 1
Rotgut                 1
Pitch Battle           1
A Park for Detroit     1
Name: primary_title, Length: 136071, dtype: int64
Viewing values in col: original_title
Top 5 values:
Broken                        19
Home                          18
The Return                    17
Alone                         13
The Gift                      13
                              ..
Back to Basics                 1
I Am Santa Claus               1
Wai Guo Ren                    1
Recoletos (arriba y abajo)     1
A Park for Detroit             1
Name: original_title, Length: 137773, dtype: int64
Viewing values in col: start_year
Top 5 values:
2017    17504
2016    17272
2018    16849
2015    16243
2014    15589
2013    14709
2012    13787
2011    1290

In [105]:
#making a list of all genres

cols = list(imdb_basics_crew_ratings.columns)

In [106]:
genre_cols = cols[11:]

In [107]:
#getting a dict with genre counts

genre_count = {}
for col in genre_cols:
    count = np.sum(imdb_basics_crew_ratings[col] == 1).sum() 
    genre_count[col] = count

In [108]:
movie_details_df_modified.head()

primary_title              original_title  \
tconst                                                                   
tt0063540                        Sunghursh                   Sunghursh   
tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
tt0069049       The Other Side of the Wind  The Other Side of the Wind   
tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
tt0100275         The Wandering Soap Opera       La Telenovela Errante   

           start_year  runtime_minutes  averagerating  numvotes  \
tconst                                                            
tt0063540        2013            175.0            7.0      77.0   
tt0066787        2019            114.0            7.2      43.0   
tt0069049        2018            122.0            6.9    4517.0   
tt0069204        2018              NaN            6.1      13.0   
tt0100275        2017             80.0            6.5     119.0   

                     directors                                  writers  \
tconst                                                                    
tt0063540            nm0712540  nm0023551,nm1194313,nm0347899,nm1391276   
tt0066787            nm0002411                                      NaN   
tt0069049            nm0000080                      nm0000080,nm0462648   
tt0069204            nm0611531                                nm0347899   
tt0100275  nm0765384,nm0749914                      nm1360635,nm0749914   

           Crime  War  ...  Music  Action  Musical  Western  Game-Show  \
tconst                 ...                                               
tt0063540    1.0  0.0  ...    0.0     1.0      0.0      0.0        0.0   
tt0066787    0.0  0.0  ...    0.0     0.0      0.0      0.0        0.0   
tt0069049    0.0  0.0  ...    0.0     0.0      0.0      0.0        0.0   
tt0069204    0.0  0.0  ...    0.0     0.0      0.0      0.0        0.0   
tt0100275    0.0  0.0  ...    0.0     0.0      0.0      0.0        0.0   

           Mystery  Sport  Comedy  Documentary  Adult  
tconst                                                 
tt0063540      0.0    0.0     0.0          0.0    0.0  
tt0066787      0.0    0.0     0.0          0.0    0.0  
tt0069049      0.0    0.0     0.0          0.0    0.0  
tt0069204      0.0    0.0     1.0          0.0    0.0  
tt0100275      0.0    0.0     1.0          0.0    0.0  

[5 rows x 36 columns]

Wonderful, now we have the genres split into usable data. Let's do something similar for writer and directors. 

In [109]:
movie_details_df_modified['directors'] = movie_details_df_modified['directors'].fillna('n/a')
movie_details_df_modified['directors'] = movie_details_df_modified['directors'].apply(lambda x: x.split(",") if x else x)
movie_details_df_modified.head()

primary_title              original_title  \
tconst                                                                   
tt0063540                        Sunghursh                   Sunghursh   
tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
tt0069049       The Other Side of the Wind  The Other Side of the Wind   
tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
tt0100275         The Wandering Soap Opera       La Telenovela Errante   

           start_year  runtime_minutes  averagerating  numvotes  \
tconst                                                            
tt0063540        2013            175.0            7.0      77.0   
tt0066787        2019            114.0            7.2      43.0   
tt0069049        2018            122.0            6.9    4517.0   
tt0069204        2018              NaN            6.1      13.0   
tt0100275        2017             80.0            6.5     119.0   

                        directors                                  writers  \
tconst                                                                       
tt0063540             [nm0712540]  nm0023551,nm1194313,nm0347899,nm1391276   
tt0066787             [nm0002411]                                      NaN   
tt0069049             [nm0000080]                      nm0000080,nm0462648   
tt0069204             [nm0611531]                                nm0347899   
tt0100275  [nm0765384, nm0749914]                      nm1360635,nm0749914   

           Crime  War  ...  Music  Action  Musical  Western  Game-Show  \
tconst                 ...                                               
tt0063540    1.0  0.0  ...    0.0     1.0      0.0      0.0        0.0   
tt0066787    0.0  0.0  ...    0.0     0.0      0.0      0.0        0.0   
tt0069049    0.0  0.0  ...    0.0     0.0      0.0      0.0        0.0   
tt0069204    0.0  0.0  ...    0.0     0.0      0.0      0.0        0.0   
tt0100275    0.0  0.0  ...    0.0     0.0      0.0      0.0        0.0   

           Mystery  Sport  Comedy  Documentary  Adult  
tconst                                                 
tt0063540      0.0    0.0     0.0          0.0    0.0  
tt0066787      0.0    0.0     0.0          0.0    0.0  
tt0069049      0.0    0.0     0.0          0.0    0.0  
tt0069204      0.0    0.0     1.0          0.0    0.0  
tt0100275      0.0    0.0     1.0          0.0    0.0  

[5 rows x 36 columns]

In [110]:
# This function takes the lists of values in the director's category and creates a new dataframe of those values
f = lambda x: 'director_{}'.format(x + 1)

director_frame = pd.DataFrame(
    movie_details_df_modified['directors'].values.tolist(),
    movie_details_df_modified.index, dtype=object
).fillna('n/a').rename(columns=f)

In [111]:
d1 = director_frame['director_1']
d2 = director_frame['director_2']
d3 = director_frame['director_3']
directors = pd.concat([d1, d2, d3], axis = 1)
directors.head()

director_1 director_2 director_3
tconst                                    
tt0063540  nm0712540        n/a        n/a
tt0066787  nm0002411        n/a        n/a
tt0069049  nm0000080        n/a        n/a
tt0069204  nm0611531        n/a        n/a
tt0100275  nm0765384  nm0749914        n/a

In [112]:
movie_details_df_modified = movie_details_df_modified.join(directors)
movie_details_df_modified.info()

<class 'pandas.core.frame.DataFrame'>
Index: 146144 entries, tt0063540 to tt9916754
Data columns (total 39 columns):
primary_title      146144 non-null object
original_title     146123 non-null object
start_year         146144 non-null int64
runtime_minutes    114405 non-null float64
averagerating      73856 non-null float64
numvotes           73856 non-null float64
directors          146144 non-null object
writers            110261 non-null object
Crime              146144 non-null float64
War                146144 non-null float64
Adventure          146144 non-null float64
Biography          146144 non-null float64
n/a                146144 non-null float64
News               146144 non-null float64
Sci-Fi             146144 non-null float64
Animation          146144 non-null float64
Fantasy            146144 non-null float64
Short              146144 non-null float64
Talk-Show          146144 non-null float64
Thriller           146144 non-null float64
Family             146144 non-n

In [113]:
movie_details_df_modified = movie_details_df_modified.drop(columns = 'directors')
movie_details_df_modified.head()

primary_title              original_title  \
tconst                                                                   
tt0063540                        Sunghursh                   Sunghursh   
tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
tt0069049       The Other Side of the Wind  The Other Side of the Wind   
tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
tt0100275         The Wandering Soap Opera       La Telenovela Errante   

           start_year  runtime_minutes  averagerating  numvotes  \
tconst                                                            
tt0063540        2013            175.0            7.0      77.0   
tt0066787        2019            114.0            7.2      43.0   
tt0069049        2018            122.0            6.9    4517.0   
tt0069204        2018              NaN            6.1      13.0   
tt0100275        2017             80.0            6.5     119.0   

                                           writers  Crime  War  Adventure  \
tconst                                                                      
tt0063540  nm0023551,nm1194313,nm0347899,nm1391276    1.0  0.0        0.0   
tt0066787                                      NaN    0.0  0.0        0.0   
tt0069049                      nm0000080,nm0462648    0.0  0.0        0.0   
tt0069204                                nm0347899    0.0  0.0        0.0   
tt0100275                      nm1360635,nm0749914    0.0  0.0        0.0   

           ...  Western  Game-Show  Mystery  Sport  Comedy  Documentary  \
tconst     ...                                                            
tt0063540  ...      0.0        0.0      0.0    0.0     0.0          0.0   
tt0066787  ...      0.0        0.0      0.0    0.0     0.0          0.0   
tt0069049  ...      0.0        0.0      0.0    0.0     0.0          0.0   
tt0069204  ...      0.0        0.0      0.0    0.0     1.0          0.0   
tt0100275  ...      0.0        0.0      0.0    0.0     1.0          0.0   

           Adult  director_1  director_2  director_3  
tconst                                                
tt0063540    0.0   nm0712540         n/a         n/a  
tt0066787    0.0   nm0002411         n/a         n/a  
tt0069049    0.0   nm0000080         n/a         n/a  
tt0069204    0.0   nm0611531         n/a         n/a  
tt0100275    0.0   nm0765384   nm0749914         n/a  

[5 rows x 38 columns]

Now let's do the same thing with writers:

In [114]:
movie_details_df_modified['writers'] = movie_details_df_modified['writers'].fillna('n/a')
movie_details_df_modified['writers'] = movie_details_df_modified['writers'].apply(lambda x: x.split(",") if x else x)
movie_details_df_modified.head()

primary_title              original_title  \
tconst                                                                   
tt0063540                        Sunghursh                   Sunghursh   
tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
tt0069049       The Other Side of the Wind  The Other Side of the Wind   
tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
tt0100275         The Wandering Soap Opera       La Telenovela Errante   

           start_year  runtime_minutes  averagerating  numvotes  \
tconst                                                            
tt0063540        2013            175.0            7.0      77.0   
tt0066787        2019            114.0            7.2      43.0   
tt0069049        2018            122.0            6.9    4517.0   
tt0069204        2018              NaN            6.1      13.0   
tt0100275        2017             80.0            6.5     119.0   

                                                writers  Crime  War  \
tconst                                                                
tt0063540  [nm0023551, nm1194313, nm0347899, nm1391276]    1.0  0.0   
tt0066787                                         [n/a]    0.0  0.0   
tt0069049                        [nm0000080, nm0462648]    0.0  0.0   
tt0069204                                   [nm0347899]    0.0  0.0   
tt0100275                        [nm1360635, nm0749914]    0.0  0.0   

           Adventure  ...  Western  Game-Show  Mystery  Sport  Comedy  \
tconst                ...                                               
tt0063540        0.0  ...      0.0        0.0      0.0    0.0     0.0   
tt0066787        0.0  ...      0.0        0.0      0.0    0.0     0.0   
tt0069049        0.0  ...      0.0        0.0      0.0    0.0     0.0   
tt0069204        0.0  ...      0.0        0.0      0.0    0.0     1.0   
tt0100275        0.0  ...      0.0        0.0      0.0    0.0     1.0   

           Documentary  Adult  director_1  director_2  director_3  
tconst                                                             
tt0063540          0.0    0.0   nm0712540         n/a         n/a  
tt0066787          0.0    0.0   nm0002411         n/a         n/a  
tt0069049          0.0    0.0   nm0000080         n/a         n/a  
tt0069204          0.0    0.0   nm0611531         n/a         n/a  
tt0100275          0.0    0.0   nm0765384   nm0749914         n/a  

[5 rows x 38 columns]

In [115]:
# This function takes the lists of values in the director's category and creates a new dataframe of those values
f = lambda x: 'writer_{}'.format(x + 1)

writers_frame = pd.DataFrame(
    movie_details_df_modified['writers'].values.tolist(),
    movie_details_df_modified.index, dtype=object
).fillna('n/a').rename(columns=f)

In [116]:
w1 = writers_frame['writer_1']
w2 = writers_frame['writer_2']
w3 = writers_frame['writer_3']
writers = pd.concat([w1, w2, w3], axis = 1)
writers.head()

writer_1   writer_2   writer_3
tconst                                    
tt0063540  nm0023551  nm1194313  nm0347899
tt0066787        n/a        n/a        n/a
tt0069049  nm0000080  nm0462648        n/a
tt0069204  nm0347899        n/a        n/a
tt0100275  nm1360635  nm0749914        n/a

In [117]:
movie_details_df_modified = movie_details_df_modified.join(writers)
movie_details_df_modified.info()

<class 'pandas.core.frame.DataFrame'>
Index: 146144 entries, tt0063540 to tt9916754
Data columns (total 41 columns):
primary_title      146144 non-null object
original_title     146123 non-null object
start_year         146144 non-null int64
runtime_minutes    114405 non-null float64
averagerating      73856 non-null float64
numvotes           73856 non-null float64
writers            146144 non-null object
Crime              146144 non-null float64
War                146144 non-null float64
Adventure          146144 non-null float64
Biography          146144 non-null float64
n/a                146144 non-null float64
News               146144 non-null float64
Sci-Fi             146144 non-null float64
Animation          146144 non-null float64
Fantasy            146144 non-null float64
Short              146144 non-null float64
Talk-Show          146144 non-null float64
Thriller           146144 non-null float64
Family             146144 non-null float64
History            146144 non-

In [118]:
movie_details_df_modified = movie_details_df_modified.drop(columns = 'writers')
movie_details_df_modified.head()

primary_title              original_title  \
tconst                                                                   
tt0063540                        Sunghursh                   Sunghursh   
tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
tt0069049       The Other Side of the Wind  The Other Side of the Wind   
tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
tt0100275         The Wandering Soap Opera       La Telenovela Errante   

           start_year  runtime_minutes  averagerating  numvotes  Crime  War  \
tconst                                                                        
tt0063540        2013            175.0            7.0      77.0    1.0  0.0   
tt0066787        2019            114.0            7.2      43.0    0.0  0.0   
tt0069049        2018            122.0            6.9    4517.0    0.0  0.0   
tt0069204        2018              NaN            6.1      13.0    0.0  0.0   
tt0100275        2017             80.0            6.5     119.0    0.0  0.0   

           Adventure  Biography  ...  Sport  Comedy  Documentary  Adult  \
tconst                           ...                                      
tt0063540        0.0        0.0  ...    0.0     0.0          0.0    0.0   
tt0066787        0.0        1.0  ...    0.0     0.0          0.0    0.0   
tt0069049        0.0        0.0  ...    0.0     0.0          0.0    0.0   
tt0069204        0.0        0.0  ...    0.0     1.0          0.0    0.0   
tt0100275        0.0        0.0  ...    0.0     1.0          0.0    0.0   

           director_1  director_2  director_3   writer_1   writer_2   writer_3  
tconst                                                                          
tt0063540   nm0712540         n/a         n/a  nm0023551  nm1194313  nm0347899  
tt0066787   nm0002411         n/a         n/a        n/a        n/a        n/a  
tt0069049   nm0000080         n/a         n/a  nm0000080  nm0462648        n/a  
tt0069204   nm0611531         n/a         n/a  nm0347899        n/a        n/a  
tt0100275   nm0765384   nm0749914         n/a  nm1360635  nm0749914        n/a  

[5 rows x 40 columns]

This is what we want our frame to look like but now I'd like to swap the n/a's we put in for Nan

In [119]:
big_chungus = movie_details_df_modified.replace('n/a', np.nan, regex=True)
big_chungus.head()

primary_title              original_title  \
tconst                                                                   
tt0063540                        Sunghursh                   Sunghursh   
tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
tt0069049       The Other Side of the Wind  The Other Side of the Wind   
tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
tt0100275         The Wandering Soap Opera       La Telenovela Errante   

           start_year  runtime_minutes  averagerating  numvotes  Crime  War  \
tconst                                                                        
tt0063540        2013            175.0            7.0      77.0    1.0  0.0   
tt0066787        2019            114.0            7.2      43.0    0.0  0.0   
tt0069049        2018            122.0            6.9    4517.0    0.0  0.0   
tt0069204        2018              NaN            6.1      13.0    0.0  0.0   
tt0100275        2017             80.0            6.5     119.0    0.0  0.0   

           Adventure  Biography  ...  Sport  Comedy  Documentary  Adult  \
tconst                           ...                                      
tt0063540        0.0        0.0  ...    0.0     0.0          0.0    0.0   
tt0066787        0.0        1.0  ...    0.0     0.0          0.0    0.0   
tt0069049        0.0        0.0  ...    0.0     0.0          0.0    0.0   
tt0069204        0.0        0.0  ...    0.0     1.0          0.0    0.0   
tt0100275        0.0        0.0  ...    0.0     1.0          0.0    0.0   

           director_1  director_2  director_3   writer_1   writer_2   writer_3  
tconst                                                                          
tt0063540   nm0712540         NaN         NaN  nm0023551  nm1194313  nm0347899  
tt0066787   nm0002411         NaN         NaN        NaN        NaN        NaN  
tt0069049   nm0000080         NaN         NaN  nm0000080  nm0462648        NaN  
tt0069204   nm0611531         NaN         NaN  nm0347899        NaN        NaN  
tt0100275   nm0765384   nm0749914         NaN  nm1360635  nm0749914        NaN  

[5 rows x 40 columns]

Looks like we have what we want before cleaning - now we can export this as a csv:

In [74]:
#big_chungus.to_csv('imdb_big_chungus_original.csv')

Now the database looks like this: 

![title](df-2.png)

In a separate workbook we've merged budget information with the imdb tconst variable, so let's add that info to BIG CHUNGUS.

In [128]:
budget_chungus = pd.read_csv('./etl_csvs/budget_chungus.csv')

In [129]:
#remove columns we don't want added to big chungus
#budget_chungus = budget_chungus.drop(columns=['id','release_date','movie', 'movie_name'])
budget_chungus = budget_chungus[['tconst', 'production_budget', 'domestic_gross','worldwide_gross', 'release_date' ]].copy()
budget_chungus.head()

tconst  production_budget  domestic_gross  worldwide_gross release_date
0        NaN               1100          181041           181041   2005-08-05
1  tt2107644               1400               0                0   2015-09-29
2        NaN               5000            1338             1338   2005-07-13
3        NaN               6000           48482           240495   1999-04-02
4        NaN               7000             900              900   2004-12-31

In [148]:
big_budge_chungus = pd.merge(big_chungus, budget_chungus, on ='tconst', how = 'left')
big_budge_chungus.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 146144 entries, 0 to 146143
Data columns (total 45 columns):
tconst               146144 non-null object
primary_title        146144 non-null object
original_title       146123 non-null object
start_year           146144 non-null int64
runtime_minutes      114405 non-null float64
averagerating        73856 non-null float64
numvotes             73856 non-null float64
Crime                146144 non-null float64
War                  146144 non-null float64
Adventure            146144 non-null float64
Biography            146144 non-null float64
n/a                  146144 non-null float64
News                 146144 non-null float64
Sci-Fi               146144 non-null float64
Animation            146144 non-null float64
Fantasy              146144 non-null float64
Short                146144 non-null float64
Talk-Show            146144 non-null float64
Thriller             146144 non-null float64
Family               146144 non-null floa

In [149]:
#Looks like our datetime conversion got lost - let's fix that:
big_budge_chungus['release_date'] = pd.to_datetime(big_budge_chungus['release_date'])
big_budge_chungus['release_date'].dtype
big_budge_chungus.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 146144 entries, 0 to 146143
Data columns (total 45 columns):
tconst               146144 non-null object
primary_title        146144 non-null object
original_title       146123 non-null object
start_year           146144 non-null int64
runtime_minutes      114405 non-null float64
averagerating        73856 non-null float64
numvotes             73856 non-null float64
Crime                146144 non-null float64
War                  146144 non-null float64
Adventure            146144 non-null float64
Biography            146144 non-null float64
n/a                  146144 non-null float64
News                 146144 non-null float64
Sci-Fi               146144 non-null float64
Animation            146144 non-null float64
Fantasy              146144 non-null float64
Short                146144 non-null float64
Talk-Show            146144 non-null float64
Thriller             146144 non-null float64
Family               146144 non-null floa

In [150]:
#remove 'n/a' values
big_budge_chungus = big_budge_chungus.replace('n/a', np.nan, regex=True)
big_budge_chungus.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 146144 entries, 0 to 146143
Data columns (total 45 columns):
tconst               146144 non-null object
primary_title        146144 non-null object
original_title       146123 non-null object
start_year           146144 non-null int64
runtime_minutes      114405 non-null float64
averagerating        73856 non-null float64
numvotes             73856 non-null float64
Crime                146144 non-null float64
War                  146144 non-null float64
Adventure            146144 non-null float64
Biography            146144 non-null float64
n/a                  146144 non-null float64
News                 146144 non-null float64
Sci-Fi               146144 non-null float64
Animation            146144 non-null float64
Fantasy              146144 non-null float64
Short                146144 non-null float64
Talk-Show            146144 non-null float64
Thriller             146144 non-null float64
Family               146144 non-null floa

In [151]:
#Okay, let's send this rendition of big chungus to a csv.
big_budge_chungus.to_csv('./etl_csvs/big_chungus.csv')